include/mega/megaclient.h

```diff
@@ -1373,6 +1373,15 @@ class MEGA_API MegaClient
     // maps node handle to public handle
     std::map<handle, handle> mPublicLinks;
 
+    // temporal nodeHandle for uploads from App
+    handle mCurrUploadId;
+
+    // generates a temporal nodeHandle for uploads from App
+    handle nextUploadId();
+
+    // maps temporal nodeHandle (used for uploads from App) to definitive nodeHandle (filled by upload completion)
+    std::map<handle, handle> mTempHandleToNodeHandle;
```

Comment:
I don't think we need this map - the NewNodes now have mAddedHandle so we know the handle created for a particular folder - you can store a parallel vector with info about each node created - so get the mAddedHandle from nn[i], and lookup yourDetails[i] to figure out which files to upload to that folder, in the completion function.